### PR TITLE
Make Symbolics.jl compat a lower bound and add doctests

### PR DESCRIPTION
**Pull request by abhro**
Using `Symbolics = "4"` compat forces the package to use an older version of Latexify, which uses Formatting.jl. Formatting.jl is archived and should be replaced with Format.jl, which is done in newer versions of Latexify.jl/Symbolics.jl.

### CODE REVIEW

**Comment by abhro** [comment]
Did not realize that committing to the same branch also adds it to the pull request. These commits probably shouldn't be merged in the same pull, but it might be messier to separate them.

**Comment by jishnub** [comment]
Thanks, this looks good overall. I'd just suggest using a more restrictive compat bound on Symbolics.

File: Project.toml [hunk 1]
```diff
@@ -13,7 +13,7 @@ OffsetArrays = "1"
 PerformanceTestTools = "0.1"
 QuadGK = "2"
 SpecialFunctions = "1, 2"
-Symbolics = "4"
+Symbolics = ">= 4"
```
Comment (jishnub) [inline Project.toml:16]:
```suggestion
Symbolics = "4, 5"
```

File: test/Project.toml [hunk 2]
```diff
@@ -14,4 +15,4 @@ Aqua = "0.5"
 HyperDualNumbers = "4"
 PerformanceTestTools = "0.1"
 QuadGK = "2"
-Symbolics = "4"
+Symbolics = ">= 4"
```
Comment (jishnub) [inline test/Project.toml:18]:
```suggestion
Symbolics = "4, 5"
```

**Comment by jishnub** [comment]
Perhaps we should carry out the doctest checks only on v1.10

**Comment by abhro** [comment]
Hmm. Seems you're right. Would it be better or worse to add an override in the REPL examples to control the number of digits to display? It might also help demonstrate numerical errors?

**Comment by jishnub** [comment]
I think that might be overkill. In any case, doctests are only really guaranteed to work on specific versions.

**Comment by abhro** [comment]
Oh. I did not know that. Okay then!

File: test/test_threaded.jl [hunk 3]
```diff
@@ -436,3 +437,13 @@ end
...
 end
+
+@testset "doctests" begin
+    DocMeta.setdocmeta!(
+        LegendrePolynomials,
+        :DocTestSetup,
+        :(using LegendrePolynomials);
+        recursive = true
+    )
+    doctest(LegendrePolynomials)
```
Comment (jishnub) [inline test/test_threaded.jl:448]:
```suggestion
    if v"1.10.2" <= VERSION < v"1.11.0-"
        doctest(LegendrePolynomials)
    end
```

**Comment by jishnub** [comment]
Could you also update the compat bound in `docs/Project.toml`?

**Comment by jishnub** [comment]
Also, I think we should just use `Symbolics="5"` everywhere instead of `"4, 5"`. This is because we're altering the docstrings to make them compatible with v5, and they won't be compatible with v4 anymore.

File: Project.toml [hunk 4]
```diff
@@ -13,7 +13,7 @@ OffsetArrays = "1"
 PerformanceTestTools = "0.1"
 QuadGK = "2"
 SpecialFunctions = "1, 2"
-Symbolics = "4"
+Symbolics = "4, 5"
```
Comment (jishnub) [inline Project.toml:16]:
```suggestion
Symbolics = "5"
```

File: test/Project.toml [hunk 5]
```diff
@@ -14,4 +15,4 @@ Aqua = "0.5"
 HyperDualNumbers = "4"
 PerformanceTestTools = "0.1"
 QuadGK = "2"
-Symbolics = "4"
+Symbolics = "4, 5"
```
Comment (jishnub) [inline test/Project.toml:18]:
```suggestion
Symbolics = "5"
```

**Pull request merged**